ethereum/contracts/Governance.sol

```diff
@@ -37,7 +37,8 @@ abstract contract Governance is GovernanceStructs, Messages, Setters, ERC1967Upg
         require(upgrade.module == module, "Invalid Module");
 
         // Verify the VAA is for this chain
-        require(upgrade.chain == chainId(), "Invalid Chain");
+        uint16 chainId = chainId();
+        require(upgrade.chain == chainId && chainId != type(uint16).max, "Invalid Chain");
```

Comment:
if we already revert here... why don't we just revert in `chainId()` directly? and instead of checking for `chainId() == type(uint16).max` in `submitRecoverChainId` we require `evmChainId() != block.chainid` there.

Reply:
I would prefer to not revert in the `chainId()` getter and instead let the caller revert. For example, we don't want to revert here https://github.com/wormhole-foundation/wormhole/blob/8f094b4d8b9f97c9b9b8490d8e9b222b1b32256b/ethereum/contracts/Governance.sol#L91 (in case we have to update the guardian set on a forked chain) and I'd rather not read the chainId variable directly there (instead of through the getter). What do you think?

Reply:
Thought more about it, and `chainId()` doesn't even need to return `type(uint16).max` for "bad forks". We can simplify by checking if `evmChainId() != block.chainid` in the appropriate places. This simplifies the code and makes it easier to understand IMO.